contracts/Template.sol

```diff
@@ -50,76 +48,80 @@ contract TemplateBase is APMNamehash {
 
         return base;
     }
+
+    function installApp(Kernel dao, bytes32 appId) internal returns (address) {
+        address instance = address(dao.newAppInstance(appId, latestVersionAppBase(appId)));
+        emit InstalledApp(instance, appId);
+        return instance;
+    }
+
+    function installDefaultApp(Kernel dao, bytes32 appId) internal returns (address) {
+        address instance = address(dao.newAppInstance(appId, latestVersionAppBase(appId), new bytes(0), true));
+        emit InstalledApp(instance, appId);
+        return instance;
+    }
 }
 
 
 contract Template is TemplateBase {
-    using SafeERC20 for ERC20;
-    MiniMeTokenFactory tokenFactory;
 
     uint64 constant PCT = 10 ** 16;
     address constant ANY_ENTITY = address(-1);
 
+    bytes32 internal CONVICTION_VOTING_APP_ID = keccak256(abi.encodePacked(apmNamehash("open"), keccak256("conviction-voting")));
+    bytes32 internal TOKEN_MANAGER_APP_ID = apmNamehash("token-manager");
+    bytes32 internal VAULT_APP_ID = apmNamehash("vault");
+
+    MiniMeTokenFactory tokenFactory;
+
     constructor(ENS ens) TemplateBase(DAOFactory(0), ens) public {
         tokenFactory = new MiniMeTokenFactory();
     }
 
-    function newInstance() public {
+    function newInstance(address requestToken) public {
         Kernel dao = fac.newDAO(this);
         ACL acl = ACL(dao.acl());
         acl.createPermission(this, dao, dao.APP_MANAGER_ROLE(), this);
+        address root = msg.sender;
 
-        address account2 = 0x8401Eb5ff34cc943f096A32EF3d5113FEbE8D4Eb;
-        bytes32 appId = keccak256(abi.encodePacked(apmNamehash("open"), keccak256("conviction-voting")));
-        bytes32 tokenManagerAppId = apmNamehash("token-manager");
-        bytes32 vaultAppId = apmNamehash("vault");
-
-        ConvictionVotingApp app = ConvictionVotingApp(dao.newAppInstance(appId, latestVersionAppBase(appId)));
-        TokenManager tokenManager = TokenManager(dao.newAppInstance(tokenManagerAppId, latestVersionAppBase(tokenManagerAppId)));
-        TokenManager tokenManager2 = TokenManager(dao.newAppInstance(tokenManagerAppId, latestVersionAppBase(tokenManagerAppId)));
-        Vault vault = Vault(dao.newAppInstance(vaultAppId, latestVersionAppBase(vaultAppId)));
+        ConvictionVotingApp convictionVoting = ConvictionVotingApp(installApp(dao, CONVICTION_VOTING_APP_ID));
+        TokenManager tokenManager = TokenManager(installApp(dao, TOKEN_MANAGER_APP_ID));
+        Vault vault = Vault(installDefaultApp(dao, VAULT_APP_ID));
 
-        MiniMeToken token = tokenFactory.createCloneToken(MiniMeToken(0), 0, "App token", 0, "APP", true);
-        MiniMeToken token2 = tokenFactory.createCloneToken(MiniMeToken(0), 0, "Fake DAI", 18, "DAI", true);
-        token.changeController(tokenManager);
-        token2.changeController(tokenManager2);
+        //stakeToken
+        MiniMeToken stakeToken = tokenFactory.createCloneToken(MiniMeToken(0), 0, "App token", 0, "APP", true);
```

Comment:
I think we could set the stake token name, symbol, and decimals as template parameters.